cmd/osm-controller/certificates.go

```diff
@@ -75,57 +79,47 @@ func getTresorOSMCertificateManager(kubeClient kubernetes.Interface, enableDebug
 	return certManager, certManager, nil
 }
 
-func getCertFromKubernetes(kubeClient kubernetes.Interface, namespace, secretName string) certificate.Certificater {
-	secrets, err := kubeClient.CoreV1().Secrets(namespace).List(context.Background(), metav1.ListOptions{})
-	if err != nil {
-		log.Error().Err(err).Msgf("Error listing secrets in namespace %q", namespace)
-	}
-	found := false
-	for _, secret := range secrets.Items {
-		if secret.Name == secretName {
-			found = true
-			break
-		}
-	}
-
-	if !found {
-		return nil
-	}
-
+// getCertFromKubernetes returns a Certificater type corresponding to the root certificate.
+// The function returns an error only if a secret is found with invalid data.
+func getCertFromKubernetes(kubeClient kubernetes.Interface, namespace, secretName string) (certificate.Certificater, error) {
 	rootCertSecret, err := kubeClient.CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
 	if err != nil {
-		log.Warn().Msgf("Error retrieving root certificate rootCertSecret %q from namespace %q; Will create a new one", secretName, osmNamespace)
-		return nil
+		// It is okay for this secret to be missing, in which case a new CA will be created along with a k8s secret
+		log.Debug().Msgf("Could not retrieve root certificate secret %q from namespace %q", secretName, osmNamespace)
+		return nil, nil
```

Comment:
```suggestion
		return nil, err
```

Reply:
Why don't you want to error in this scenario as well?

Reply:
This isn't an error, check the function description and existing behavior.

Reply:
Also comment on L87.

Reply:
The reason this isn't an error because the Get() api is used to only check if the secret exists, if it doesn't its ok (no need to log an error because its not an error from the controller's perspective). This is more of a helper function to parse the root certificate from the secret, the caller decides how to use the returned root cert (and creates a root cert secret if it doesn't exist).